sfa_dash/blueprints/main.py

```diff
@@ -23,6 +25,137 @@ class SingleObjectView(DataDashView):
     """
     template = 'data/asset.html'
 
+    def __init__(self, data_type):
```

Comment:
is it really necessary to separate this out. Seems like a little careful handling of getting site/aggregate_id would avoid the extra copy pasta

Reply:
You're right. I think before I arrived at some of the refactoring I did this seemed reasonable but this could probably all be combined again.